File: frontend/src/api/stats.ts
import request from './request'

export interface DashboardTask {
  id: string
  name: string
  task_type: string
  status: string
  progress: number
  result: Record<string, unknown> | null
  error_message: string | null
  created_at: string
  updated_at: string
}

export interface DashboardOverview {
  dataset_count: number
  image_count: number
  annotated_image_count: number
  model_count: number
  training_task_count: number
  running_task_count: number
  pending_review_count: number
  rejected_review_count: number
  completed_review_count: number
  recent_tasks: DashboardTask[]
}

export function getDashboardOverview() {
  return request.get<never, DashboardOverview>('/stats/overview')
}

File: backend/app/schemas/stats.py
from __future__ import annotations

import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel


class DashboardTaskResponse(BaseModel):
    id: uuid.UUID
    name: str
    task_type: str
    status: str
    progress: int
    result: dict[str, Any] | None
    error_message: str | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class DashboardOverviewResponse(BaseModel):
    dataset_count: int
    image_count: int
    annotated_image_count: int
    model_count: int
    training_task_count: int
    running_task_count: int
    pending_review_count: int
    rejected_review_count: int
    completed_review_count: int
    recent_tasks: list[DashboardTaskResponse]


class AnnotationStatsResponse(BaseModel):
    image_count: int
    annotated_image_count: int
    pending_review_count: int
    rejected_review_count: int
    completed_review_count: int
